/* Copyright 2023 The FedLearner Authors. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package fedlearner_webconsole.proto;

message CleanupPayload {
   repeated string paths = 1;
}

message CleanupParameter {
   int64 resource_id = 1;
   // Ref to ResourceType enum, here is the name value.
   string resource_type = 2;
   // Timestamp in seconds
   int64 target_start_at = 3;
   CleanupPayload payload = 4;
}
 
message CleanupPb {
   int64 id = 1;
   // Ref to CleanupStatus enum, here is the enum name.
   string state = 2;
   // Timestamp in seconds
   int64 target_start_at = 3;
   // Timestamp in seconds
   int64 completed_at = 4;
   int64 resource_id = 5;
   // Ref to ResourceType enum, here is the enum name.
   string resource_type = 6;
   CleanupPayload payload = 7;
   // Timestamp in seconds
   int64 created_at = 8;
   // Timestamp in seconds
   int64 updated_at = 9;
}